src/Clock/Timebox.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Clock;
+
+use Throwable;
+
+final readonly class Timebox
```

Comment:
This really feels like an early abstraction to me. You mentioned HTTP (I'd recommend a middleware) and scheduling in the console. What are the other use cases for this utility?

Reply:
There are no other current use-cases, both of the things I mentioned could be solved locally in said places, but it felt like a good place to start the conversation about handling dates overall and unifying how we approach them, as well as a reminder to handle that while implementing the auth.

If this gets merged we can simplify how `Every` and `Interval` are implemented in tempest/console